subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -424,6 +424,60 @@ static int cmd_wifi_ap_disable(const struct shell *sh, size_t argc,
 	return 0;
 }
 
+
+static int cmd_wifi_reg_domain(const struct shell *sh, size_t argc,
+			       char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	struct wifi_reg_domain regd;
+	bool force = false;
+	int ret;
+
+	if (argc == 1) {
+		regd.oper = WIFI_MGMT_GET;
+	} else if (argc == 2) {
+		regd.oper = WIFI_MGMT_SET;
+		if (strlen(argv[1]) != 2) {
+			shell_fprintf(sh, SHELL_WARNING, "Invalid reg domain\n");
+			return -ENOEXEC;
+		}
+
+		/* Two letter country code with special case of 00 for WORLD */
+		if ((argv[1][0] < 'A' || argv[1][0] > 'Z') ||
+		    (argv[1][1] < 'A' || argv[1][1] > 'Z') ||
+			(argv[1][0] == '0' || argv[1][1] == '0')) {
+			shell_fprintf(sh, SHELL_WARNING, "Invalid reg domain\n");
+			return -ENOEXEC;
+		}
+		regd.country_code[0] = argv[1][0];
+		regd.country_code[1] = argv[1][1];
+	} else if (argc == 3) {
+		force = argv[2] == 1 ? true : false;
```

Comment:
Actually, the `force` variable doesn't seeem to be used anywhere. Plus, the comparison looks fishy (pointer vs int).